### PR TITLE
universe: make additional details optional to speed up query

### PR DESCRIPTION
**Pull request by guggero**
This makes the additional DB query for each grouped asset optional and hides it behind an additional gRPC/REST parameter.

### CODE REVIEW

File: taprpc/universerpc/universe.swagger.json [hunk 1]
```diff
@@ -791,6 +791,15 @@
             }
           }
         },
+        "parameters": [
+          {
+            "name": "with_amounts_by_id",
+            "description": "TODO(roasbeef): filter by asset ID, etc?.",
```
Comment (dstadulis) [inline taprpc/universerpc/universe.swagger.json:797]: Is this TODO supposed to be here?

Comment (guggero) [reply taprpc/universerpc/universe.swagger.json]: Nice catch! The comment was there previously but wasn't mapped to a specific field. Updated that.

**Review by jharveyb — APPROVED** [review]
LGTM! I see the linter issue is fixed in #609.

Do we want to update the CLI to maintain the old behavior, or expose this setting? In favor of just keeping the old behavior imo.

https://github.com/lightninglabs/taproot-assets/blob/c7d57c597dfaadf0732a9ffb4bf63152c427f8f8/cmd/tapcli/universe.go#L157

**Comment by Roasbeef** [comment]
Was this guided optimization via `EXPLAIN ANALYZE`, or something else? Do you have that dump we can use to add more detail to the commit body?

**Comment by Roasbeef** [comment]
Ah reading more now, I see it's to just leave off some extra queries if/when only the roots are needed. 

I think we'll want to have the REST interface display by default, or just make sure the new query param there is documented.

**Comment by guggero** [comment]
> I think we'll want to have the REST interface display by default, or just make sure the new query param there is documented.

Yeah, I should've added more info here, sorry.
So this extra loop to fetch all assets within a group was a user experience improvement added in https://github.com/lightninglabs/taproot-assets/pull/492 to better understand the output of the roots call.
But because this does an extra DB call for each group, it is very expensive. So we definitely should **NOT** turn this on for universe sync _and_ the REST call. I checked with @jamaljsr and we don't use the extra information on Terminal Web.
The extra info is only required in Polar where Jamal is going to add the query parameter (but there it will just query the local universe).

But I think we do want the CLI to remain at the previous behavior, so I'm going to fix that.

File: cmd/tapcli/universe.go [hunk 2]
```diff
@@ -70,6 +71,11 @@ var universeRootsCommand = cli.Command{
 				"either 'issuance' or 'transfer'",
 			Value: universe.ProofTypeIssuance.String(),
 		},
+		cli.BoolFlag{
+			Name: skipAmountsByIdName,
+			Usage: "skip showing the amounts by ID for grouped " +
+				"assets to optimize response size and speed",
+		},
 	},
```
Comment (ffranr) [inline cmd/tapcli/universe.go:79]: Doesn't this default to `false`? And so we wont skip? I think our intent is to skip by default. I think the flag should capture the positive case here: `withAmountsById`.

Comment (ffranr) [reply cmd/tapcli/universe.go:79]: > But I think we do want the CLI to remain at the previous behavior, so I'm going to fix that.

I see this comment from you now.

**Review by ffranr — APPROVED** [review]
Looks good :fireworks:

**Review by Roasbeef — APPROVED** [review]
LGTM 🐌

**Pull request merged**